applications/crossbar/src/modules/cb_callflows.erl

```diff
@@ -387,6 +385,48 @@ validate_callflow_elements(Context) ->
 %%--------------------------------------------------------------------
 %% @private
 %% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec validate_number_ownership(cb_context:context()) -> cb_context:context().
+-spec validate_number_ownership(ne_binaries(), cb_context:context()) ->
+                                       cb_context:context().
+-spec validate_number_ownership(knm_number:knm_number()
+                               ,ne_binaries()
+                               ,cb_context:context()) -> cb_context:context().
+validate_number_ownership(Context) ->
+    NewNums = kz_json:get_value(<<"numbers">>, cb_context:doc(Context), []),
+
+    Assigned =  [Num
+                 || Num <- NewNums,
+                    Num =/= <<"undefined">>
+                ],
+
+    validate_number_ownership(Assigned, Context).
+
+validate_number_ownership([], Context) -> Context;
+validate_number_ownership([Number|Numbers], Context) ->
+    case knm_number:get(Number) of
+        {'ok', NumberObj} ->
+            validate_number_ownership(NumberObj, Numbers, Context);
+        _ -> validate_number_ownership(Numbers, Context)
+    end.
+
+validate_number_ownership(NumberObj, Numbers, Context) ->
+    AuthBy = cb_context:auth_account_id(Context),
+    PhoneNumber = knm_number:phone_number(NumberObj),
+    AssignedTo = knm_phone_number:assigned_to(PhoneNumber),
+    case kz_util:is_in_account_hierarchy(AuthBy, AssignedTo, 'true') of
```

Comment:
Hey since this seems to only need the assigned_to field of a number, can you use the view `numbers/assigned_to` instead of multiple knm_number:get calls? That way you can also group numbers by AssignedTo and do this AuthBy check once per group of numbers.